frame/transaction-payment/asset-tx-payment/src/lib.rs

```diff
@@ -119,6 +119,12 @@ pub mod pallet {
 		type Fungibles: Balanced<Self::AccountId>;
 		/// The actual transaction charging logic that charges the fees.
 		type OnChargeAssetTransaction: OnChargeAssetTransaction<Self>;
+		/// Type that implements the `BalanceConversion` trait.
+		type BalanceToAsset: BalanceConversion<
+			BalanceOf<Self>,
+			AssetIdOf<Self>,
+			AssetBalanceOf<Self>,
+		>;
```

Comment:
I need to look into this a bit more. Not saying it's wrong, but need to look at this whole pallet.

With the DEX pallet, I want this Asset Tx Payment pallet to be more configurable. We want to be able to configure it like,

- "Convert the native fee balance to an asset amount, using ratio of minimum balances (`AssetBalanceOf`), and make the payment in the `AssetId`"
- "Calculate the fee in the native asset, and then swap `AssetId` in the Dex, paying the collator in `BalanceOf`".

Perhaps should log this as a separate issue and do a refactor of this pallet that could have breaking changes to this interface.

Reply:
Yes this is a separate issue. 

> "Convert the native fee balance to an asset amount, using ratio of minimum balances (AssetBalanceOf), and make the payment in the AssetId"

Especially this would require a new signed extension, because it works in a quite different way than the current implementation. Currently you pay in an asset and you want to start swapping assets before. I'm not even sure if this needs to be part of `asset-tx-payment` and isn't more its own logic/signed extension in the dex. But yeah, new issue! :D